lib/event/emitter.js

```diff
@@ -39,7 +39,12 @@ EventEmitter.prototype.emit = function(type) {
   if (!listeners) {
     return;
   }
-  var args = Array.prototype.slice.call(arguments, 1);
+  // We need to copy arguments to local array for using it in another function call.
+  // The only way js engine (like V8) will be able to optimize it.
+  var args = [];
```

Comment:
```new Array(arguments.length)``` is the best solution for browsers based on latest version on V8 (Chrome, Opera, etc), but for some browsers (old IE8, IE9, old V8 based browsers, maybe Firefox - I'm not sure now) ```args = [] ``` and ```args.push() ``` is faster. So I think, ```args = [] ``` - is good, is middle way - newest browsers are fast, older browsers - work as quickly as possible